sdk/tables/azure-data-tables/src/samples/java/TableServiceAsyncClientCodeSnippets.java

```diff
@@ -0,0 +1,204 @@
+            tableEntity.addProperty("Price", "5");
+            Mono<Void> updateEntityMono = tableAsyncClient.upsertEntity(UpdateMode.Merge, tableEntity);
+            return updateEntityMono;
+        }).subscribe(Void -> {
+            logger.info("Update Entity Successful.");
+        }, error -> {
+            logger.error("There was an error updating the Entity. Error: " + error);
+        });
+    }
+
+    /**
+     * update entity code snippet
+     */
+    private void update() {
+
+        // Build service client
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildAsyncClient();
+
+        TableAsyncClient tableAsyncClient = tableServiceAsyncClient.getTableAsyncClient("OfficeSupplies");
+        QueryOptions queryOptions = new QueryOptions();
+        queryOptions.setFilter("RowKey eq crayolaMarkers");
+
+        tableAsyncClient.queryEntity(queryOptions).flatMap(tableEntity -> {
+            logger.info("Table Entity: " + tableEntity);
+            tableEntity.addProperty("Price", "5");
+            Mono<Void> updateEntityMono = tableAsyncClient.updateEntity(UpdateMode.Replace, tableEntity);
+            return updateEntityMono;
+        }).subscribe(Void -> {
+            logger.info("Update Entity Successful.");
+        }, error -> {
+            logger.error("There was an error updating the Entity. Error: " + error);
+        });
+    }
+
+    /**
+     * query entity code snippet
+     */
+    private void queryEntities() {
+
+        // Build service client
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildAsyncClient();
+
+        TableAsyncClient tableAsyncClient = tableServiceAsyncClient.getTableAsyncClient("OfficeSupplies");
+        QueryOptions queryOptions = new QueryOptions();
+        queryOptions.setFilter("Product eq markers");
+        queryOptions.setSelect("Seller, Price");
+
+        tableAsyncClient.queryEntity(queryOptions).subscribe(tableEntity -> {
+            logger.info("Table Entity: " + tableEntity);
+        }, error -> {
+            logger.error("There was an error querying the table. Error: " + error);
+        });
+    }
+
+    /**
+     * checks to see if an entity exists code snippet
+     */
+    private void existsEntity() {
+
+        // Build service client
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildAsyncClient();
+
+        TableAsyncClient tableAsyncClient = tableServiceAsyncClient.getTableAsyncClient("OfficeSupplies");
+
+        tableAsyncClient.queryEntitiesWithPartitionAndRowKey("crayolaMarkers", "markers")
+            .subscribe(tableEntity -> {
+            logger.info("Table Entity exists: " + tableEntity);
```

Comment:
This returns a paged collection. So, you might have to check each `tableEntity` to find out if the one that you are testing for existence actually is in the returned collection.

Reply:
updated using get() method.